pkg/skaffold/kubernetes/portforward/resource_forwarder.go

```diff
@@ -67,14 +67,23 @@ func NewUserDefinedForwarder(entryManager *EntryManager, userDefinedResources []
 func (p *ResourceForwarder) Start(ctx context.Context, namespaces []string) error {
 	if len(namespaces) == 1 {
 		for _, pf := range p.userDefinedResources {
-			if pf.Namespace == "" {
+			if pf.Namespace != "" {
+				err := applyNamespaceWithTemplate(pf)
+				if err != nil {
+					return err
+				}
+			} else {
 				pf.Namespace = namespaces[0]
 			}
 		}
 	} else {
 		var validResources []*latestV1.PortForwardResource
 		for _, pf := range p.userDefinedResources {
 			if pf.Namespace != "" {
+				err := applyNamespaceWithTemplate(pf)
+				if err != nil {
```

Comment:
Can inline this and save a line
```suggestion
				if err := applyNamespaceWithTemplate(pf); err != nil {
```

Reply:
@briandealwis done!